src/lib/tls/tls_extensions.cpp

```diff
@@ -761,7 +764,27 @@ PSK_Key_Exchange_Modes::PSK_Key_Exchange_Modes(TLS_Data_Reader& reader, uint16_t
 }
 
 std::vector<uint8_t> Certificate_Authorities::serialize(Connection_Side) const {
-   throw Not_Implemented("serializing Certificate_Authorities is NYI");
+   std::vector<uint8_t> bytes(2);
+   auto encoder = DER_Encoder(bytes);
+
+   for (auto& dn : m_distinguished_names) {
+       auto sz_idx = bytes.size();
+       bytes.emplace_back();
+       bytes.emplace_back();
+       auto old_sz = bytes.size();
+       dn.encode_into(encoder);
+       auto sz = static_cast<uint16_t>(bytes.size() - old_sz);
+
+       bytes[sz_idx] = get_byte<0>(sz);
+       bytes[sz_idx+1] = get_byte<1>(sz);
+   }
+
+   BOTAN_ASSERT_NOMSG(bytes.size() >= 2 && bytes.size() <= std::numeric_limits<uint16_t>::max());
```

Comment:
`append_tls_length_value()` also checks the size limits and throws if things don't fit. Once you moved your application to use it, this assertion should be removed.

As a general remark: We use assertions to validate things that should never happen in any typical use case. However, this might be triggered by an application that configured too many trusted DNs. It should therefore not be an assertion but a sanity check.